--- src/rendervulkan.cpp	2024-10-05 08:39:13.515975808 -0400
+++ src/rendervulkan.cpp	2024-10-05 08:40:56.954189908 -0400
@@ -2799,7 +2799,7 @@
 	}
 	else
 	{
-		if ( !GetBackend()->SupportsFormat( drmFormat ) )
+		if ( GetBackend()->UsesModifiers() && !GetBackend()->SupportsFormat( drmFormat ) )
 			return false;
 
 		wlr_drm_format_set_add( &sampledDRMFormats, drmFormat, DRM_FORMAT_MOD_INVALID );
