--- a/src/meson.build	2025-05-27 14:38:03.914342082 -0400
+++ b/src/meson.build	2025-05-27 14:39:06.216654161 -0400
@@ -124,6 +124,7 @@
 
 luajit_dep = dependency( 'luajit' )
 libinput_dep = dependency('libinput', required: true)
+libudev_dep = dependency('libudev', required: true)
 
 gamescope_cpp_args = []
 if drm_dep.found()
@@ -197,7 +198,7 @@
       xkbcommon, thread_dep, sdl2_dep, wlroots_dep,
       vulkan_dep, liftoff_dep, dep_xtst, dep_xmu, cap_dep, epoll_dep, pipewire_dep, librt_dep,
       stb_dep, displayinfo_dep, openvr_dep, dep_xcursor, avif_dep, dep_xi,
-      libdecor_dep, eis_dep, luajit_dep, libinput_dep, libsystemd_dep,
+      libdecor_dep, eis_dep, luajit_dep, libinput_dep, libudev_dep, libsystemd_dep,
     ],
     install: true,
     cpp_args: gamescope_cpp_args,
