diff --git a/samples/CMakeLists.txt b/samples/CMakeLists.txt
index 545ff1a..dacb2ba 100644
--- a/samples/CMakeLists.txt
+++ b/samples/CMakeLists.txt
@@ -137,35 +137,44 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     list(APPEND SDL2_LIBRARIES ${${lib}_LIBRARY})
   endforeach()
 else()
-  foreach(lib ${SDL_REQUIRED_LIBRARIES})
-    find_library(${lib}_LIBRARY
-      NAMES
+  if (USE_SYSTEM_SDL2)
+    find_package(${SDL_REQUIRED_LIBRARIES} REQUIRED)
+  else()
+    foreach(lib ${SDL_REQUIRED_LIBRARIES})
+      find_library(${lib}_LIBRARY
+        NAMES
         ${lib}
-      PATHS
-        ${THIRDPARTY_DIR}/sdl2-2.0.3/bin
-      PATH_SUFFIXES
-        linux${PLATFORM}
-        ${WINDOWS_PATH_SUFFIXES}
-    )
-    list(APPEND SDL2_LIBRARIES ${${lib}_LIBRARY})
-  endforeach()
+        PATHS
+          ${THIRDPARTY_DIR}/sdl2-2.0.3/bin
+        PATH_SUFFIXES
+          linux${PLATFORM}
+          ${WINDOWS_PATH_SUFFIXES}
+      )
+      list(APPEND SDL2_LIBRARIES ${${lib}_LIBRARY})
+    endforeach()
+  endif()
 endif()
 set(SDL2_INCLUDE_DIR ${THIRDPARTY_DIR}/sdl2-2.0.3/include)
 
 ## Vulkan
 if(NOT (${CMAKE_SYSTEM_NAME} MATCHES "Darwin"))
-  find_library(VULKAN_LIBRARY
-      NAMES
-        vulkan-1
-        vulkan
-      PATHS
-        ${THIRDPARTY_DIR}/vulkan-1.0.49.0/lib
-      PATH_SUFFIXES
-        linux${PLATFORM}
-        ${WINDOWS_PATH_SUFFIXES}
-      NO_DEFAULT_PATH
-  )
-  set(VULKAN_INCLUDE_DIR ${THIRDPARTY_DIR}/vulkan-1.0.49.0/include)
+  if (USE_SYSTEM_VULKAN)
+    find_package(Vulkan REQUIRED)
+    list(APPEND EXTRA_LIBS ${Vulkan_LIBRARIES})
+  else()
+    find_library(VULKAN_LIBRARY
+        NAMES
+          vulkan-1
+          vulkan
+        PATHS
+          ${THIRDPARTY_DIR}/vulkan-1.0.49.0/lib
+        PATH_SUFFIXES
+          linux${PLATFORM}
+          ${WINDOWS_PATH_SUFFIXES}
+        NO_DEFAULT_PATH
+    )
+    set(VULKAN_INCLUDE_DIR ${THIRDPARTY_DIR}/vulkan-1.0.49.0/include)
+  endif()
 endif()
 
 
