diff -Naur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2023-10-24 19:58:37.076421394 -0700
+++ b/src/CMakeLists.txt	2023-10-24 20:00:26.807622046 -0700
@@ -78,6 +78,7 @@
 # Build the library.
 if(BUILD_SHARED)
 	add_library(${LIBNAME} SHARED ${SOURCE_FILES})
+	set_target_properties(${LIBNAME} PROPERTIES SOVERSION 2.0.0)
 elseif(BUILD_FRAMEWORK)
 	set( LIBNAME "OpenVR" )
 	add_library( ${LIBNAME} 
